source/server/hot_restarting_child.cc

```diff
@@ -80,19 +80,31 @@ void HotRestartingChild::sendParentTerminateRequest() {
   wrapped_request.mutable_request()->mutable_terminate();
   sendHotRestartMessage(parent_address_, wrapped_request);
   parent_terminated_ = true;
-  // Once setting parent_terminated_ == true, we can send no more hot restart RPCs, and therefore
-  // receive no more responses, including stats. So, now safe to forget our stat transferral state.
+
+  // Note that we the 'generation' counter needs to retain the contribution
+  // from the parent....
+  stat_merger_->retainParentGaugeValue(hot_restart_generation_stat_name_);
+
+  // Once setting parent_terminated_ == true, we can send no more hot restart
+  // RPCs, and therefore receive no more responses, including stats. We should
+  // also zero out parent-stat contribution to child gauges (see
+  // https://github.com/envoyproxy/envoy/issues/10806).
+  stat_merger_->removeParentContributionToGauges();
```

Comment:
nit: can we just do this in the stat merger destructor? I don't feel strongly about it but seems like we won't need this method.

Reply:
yes...it changed the semantics of one of the tests a bit, but it's fine.